backend/src/types/Events.ts

```diff
@@ -0,0 +1,119 @@
+import { GameCard } from './Cards';
+
+export enum EventName {
+  game_start = 'game_start',
+  shuffle_cards = 'shuffle_cards',
+  start_turn = 'start_turn',
+  end_turn = 'end_turn',
+  play_card = 'play_card',
+  discard_card = 'discard_card',
+  draw_card = 'draw_card',
+  health = 'health'
+}
+
+export enum SourceKind {
+  deck = 'deck',
+  discard = 'discard',
+  playerHand = 'playerHand',
+  playerBoard = 'playerBoard',
+}
+
+export interface GameEvent {
+  name: EventName;
+  data: any;
+}
+
+export class GameStartEvent implements GameEvent {
+  name = EventName.game_start;
+  data = null;
+}
+
+export class ShuffleCardsEvent implements GameEvent {
+  name = EventName.shuffle_cards;
+  data = null;
+}
+
+export interface StartTurnData {
+  prevPlayer: String,
+  nextPlayer: String,
+}
+
+export class StartTurnEvent implements GameEvent {
+  constructor (data: StartTurnData) {

```

Comment:
make a call to super
super(EventName.START_TURN, data)

each class won't store its own data, it'll be handled by the GameEvent class